xml/admin_ceph_upgrade.xml

```diff
@@ -179,37 +184,428 @@
+stage_prep: disabled
+stage_discovery: disabled
+stage_configure: disabled
+stage_deploy: disabled
+stage_services: disabled
+stage_remove: disabled
+</screen>
+    <para>
+     Save the file and apply the changes:
+    </para>
+<screen>&prompt.smaster;salt '*' saltutil.pillar_refresh</screen>
+   </step>
+   <step>
+    <para>
+     If you are <emphasis role="bold">not</emphasis> using container images
+     from <literal>registry.suse.com</literal> but rather the locally
+     configured registry, edit
+     <filename>/srv/pillar/ceph/stack/global.yml</filename> to inform &deepsea;
+     which &ceph; container image and registry to use. For example, to use
+     <literal>192.168.121.1:5000/my/ceph/image</literal> from an insecure
+     (HTTP) registry, add the following lines:
+    </para>
+<screen>
+ses7_container_image: 192.168.121.1:5000/my/ceph/image
+ses7_container_registries:
+  - location: 192.168.121.1:5000
+  - insecure: true
+</screen>
+    <para>
+     Save the file and apply the changes:
+    </para>
+<screen>&prompt.smaster;alt '*' saltutil.refresh_pillar</screen>
+   </step>
+   <step>
+    <para>
+     Assimilate existing configuration:
+    </para>
+<screen>&prompt.cephuser;ceph config assimilate-conf -i /etc/ceph/ceph.conf</screen>
+   </step>
+   <step>
+    <para>
+     Verify the upgrade status. Your output may differ depending on your
+     cluster configuration:
+    </para>
+<screen>
+&prompt.smaster;salt-run upgrade.status
+The newest installed software versions are:
+ ceph: ceph version 15.2.2-60-gf5864377ab (f5864377abb5549f843784c93577980aa264b9bc) octopus (stable)
+ os: SUSE Linux Enterprise Server 15 SP2
+Nodes running these software versions:
+ admin.ceph (assigned roles: master, prometheus, grafana)
+Nodes running older software versions must be upgraded in the following order:
+ 1: mon1.ceph (assigned roles: admin, mon, mgr)
+ 2: mon2.ceph (assigned roles: admin, mon, mgr)
+ 3: mon3.ceph (assigned roles: admin, mon, mgr)
+ 4: data4.ceph (assigned roles: storage, mds)
+ 5: data1.ceph (assigned roles: storage)
+ 6: data2.ceph (assigned roles: storage)
+ 7: data3.ceph (assigned roles: storage)
+ 8: data5.ceph (assigned roles: storage, rgw)
+</screen>
+   </step>
+  </procedure>
  </sect1>
  <sect1 xml:id="upgrade-mon-mgr-nodes">
-  <title>Upgrading the &mon; and &mgr; Nodes</title>
+  <title>Upgrade the MON, MGR, and OSD Nodes</title>
+
+  <para>
+   Upgrade MON, MGR, and OSD nodes subsequently, one at a time. For each of
+   them, follow these steps:
+  </para>
+
+  <procedure>
+   <step>
+    <para>
+     If the node you are upgrading is an OSD node, avoid having the OSD marked
+     out during the upgrade by running
+    </para>
+<screen>&prompt.cephuser;ceph osd add-noout <replaceable>SHORT_NODE_NAME</replaceable></screen>
+    <para>
+     Replace <replaceable>SHORT_NODE_NAME</replaceable> with the short name of
+     the node as it appears in the output of the <command>ceph osd
+     tree</command> command.
```

Comment:
Perhaps an example here would be helpful. @tbazant @tserong ?

Reply:
In my crappy toy test cluster of VMs, `ceph osd tree` prints:

```
# ceph osd tree
ID   CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
 -1         0.60405  root default                             
-11         0.11691      host data1                           
  4    hdd  0.01949          osd.4       up   1.00000  1.00000
  9    hdd  0.01949          osd.9       up   1.00000  1.00000
 13    hdd  0.01949          osd.13      up   1.00000  1.00000
 17    hdd  0.01949          osd.17      up   1.00000  1.00000
 21    hdd  0.01949          osd.21      up   1.00000  1.00000
 26    hdd  0.01949          osd.26      up   1.00000  1.00000
 -5         0.11691      host data2                           
  2    hdd  0.01949          osd.2       up   1.00000  1.00000
  5    hdd  0.01949          osd.5       up   1.00000  1.00000
 10    hdd  0.01949          osd.10      up   1.00000  1.00000
 15    hdd  0.01949          osd.15      up   1.00000  1.00000
 20    hdd  0.01949          osd.20      up   1.00000  1.00000
 25    hdd  0.01949          osd.25      up   1.00000  1.00000
 -7         0.13640      host data3                           
  3    hdd  0.01949          osd.3       up   1.00000  1.00000
  8    hdd  0.01949          osd.8       up   1.00000  1.00000
 14    hdd  0.01949          osd.14      up   1.00000  1.00000
 19    hdd  0.01949          osd.19      up   1.00000  1.00000
 24    hdd  0.01949          osd.24      up   1.00000  1.00000
 29    hdd  0.01949          osd.29      up   1.00000  1.00000
 30    hdd  0.01949          osd.30      up   1.00000  1.00000
 -9         0.11691      host data4                           
  0    hdd  0.01949          osd.0       up   1.00000  1.00000
  7    hdd  0.01949          osd.7       up   1.00000  1.00000
 11    hdd  0.01949          osd.11      up   1.00000  1.00000
 16    hdd  0.01949          osd.16      up   1.00000  1.00000
 22    hdd  0.01949          osd.22      up   1.00000  1.00000
 27    hdd  0.01949          osd.27      up   1.00000  1.00000
 -3         0.11691      host data5                           
  1    hdd  0.01949          osd.1       up   1.00000  1.00000
  6    hdd  0.01949          osd.6       up   1.00000  1.00000
 12    hdd  0.01949          osd.12      up   1.00000  1.00000
 18    hdd  0.01949          osd.18      up   1.00000  1.00000
 23    hdd  0.01949          osd.23      up   1.00000  1.00000
 28    hdd  0.01949          osd.28      up   1.00000  1.00000
```

My hosts are named data1.ceph, data2.ceph and so on, but as you can see the short names above are just data1, data2. So If I were upgrading the data1 node, I'd run `ceph osd add-noout data1`. I'm not sure that entire wall of text wants to be used as an example though ;-)